webui/src/util/tablesUtil.jsx

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import {objects} from "../lib/api";
+import {useAPI} from "../lib/hooks/api";
+import {TableType, TreeItemType} from "../constants";
+
+/**
+ * Checks whether a path is a table root and calculates its type.
+ *
+ * @param path the table path
+ * @param repo the repo for the path
+ * @param refs the ref in which the path is expected to be listed.
+ * @return the type of table if the path is a table root, and null otherwise (for paths to objects or prefixes).
+ */
+export function getTableType(path, repo, ref) {
```

Comment:
I think you can change this function's intention as it only checks whether or not the given path represents a delta table.

`function isDeltaTable(path, repo,ref) {... return true/false}`

Reply:
thought about this but chose getTableType with a thought of additional table types we support in mind. but I guess that it may be better to simplify at this point and extend to multi-table support in the future.

done